src/main/plugins/org.dita.base/xsl/preprocess/conrefImpl.xsl

```diff
@@ -34,18 +34,32 @@ See the accompanying LICENSE file for applicable license.
 
   <xsl:key name="id" match="*[@id]" use="@id"/>
 
+  <xsl:variable name="root" select="/" as="document-node()"/>
+
   <xsl:template match="/">
-    <xsl:apply-templates>
-      <xsl:with-param name="conref-ids" tunnel="yes" select="()"/>
-    </xsl:apply-templates>
+    <xsl:variable name="resolved" as="document-node()">
+      <xsl:document>
+        <xsl:apply-templates>
+          <xsl:with-param name="conref-ids" tunnel="yes" select="()"/>
+        </xsl:apply-templates>
+      </xsl:document>
+    </xsl:variable>
+    <xsl:variable name="resolved-ids" as="document-node()">
+      <xsl:document>
+        <xsl:apply-templates select="$resolved/node()" mode="id"/>
+      </xsl:document>
+    </xsl:variable>
+    <xsl:apply-templates select="$resolved-ids/node()" mode="href"/>
```

Comment:
So the processing is split into three stages.

1. Resolve conrefs recursively and output attributes in conreffed content using `conref` namespace for resolution in following steps.
2. Resolve `conref:orig-id` that are ID:s in conreffed content. If the same ID is not used in non-conreffed content, use the original ID; otherwise use `generate-id` to create a new ID. The latter option was the only output in the old code.
3. Resolve `conref:href` which checks if some link types in conreffed content should resolve to the original ID or generated ID.

Steps 2 and 3 cannot be merged because we need to know the effective ID values first, before the know what element ID to use in the links.